File: LearningLens2025/legacy_files/focused-ai-ui/lib/models/test_case.dart
class TestCase {
  final String id;
  final String name;
  final String input;
  final String expectedOutput;
  final double points;
  final bool isVisible;

  TestCase({
    required this.id,
    required this.name,
    required this.input,
    required this.expectedOutput,
    required this.points,
    this.isVisible = true,
  });

  factory TestCase.fromJson(Map<String, dynamic> json) {
    return TestCase(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      input: json['input'] ?? '',
      expectedOutput: json['expectedOutput'] ?? '',
      points: (json['points'] ?? 1.0).toDouble(),
      isVisible: json['isVisible'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'input': input,
      'expectedOutput': expectedOutput,
      'points': points,
      'isVisible': isVisible,
    };
  }
}

File: LearningLens2025/legacy_files/focused-ai-grader-service/backend/src/main/java/com/focusedai/grading/model/TestCase.java
package com.focusedai.grading.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class TestCase {
    @JsonProperty("id")
    private String id;
    
    @JsonProperty("name")
    private String name;
    
    @JsonProperty("input")
    private String input;
    
    @JsonProperty("expectedOutput")
    private String expectedOutput;
    
    @JsonProperty("points")
    private int points = 1;
    
    @JsonProperty("isVisible")
    private boolean isVisible = true;
    
    // Constructors
    public TestCase() {}
    
    // Getters and Setters (generate these in your IDE)
    // ... (include all getters and setters)
}